default/python/AI/aistate_interface.py

```diff
@@ -0,0 +1,22 @@
+aistate = None
```

Comment:
I prefer to access `aistate` directly without getter, this is one of the most usable object in a game.

May be we can rename it to shorter but steal meaningless name: `ai`. This object not only contains persistent state, but also some business logic.

Reply:
> I prefer to access aistate directly without getter, this is one of the most usable object in a game.

Are you saying that you think the computational cost of using a getter for this everywhere outweighs the benefits listed in the PR summary? If that's your concern then we could simply get it once at the beginning of every function where it is used ```aistate = get_aistate()``` instead of calling the getter every time we want to reference it. 

I have to note, though, I would not be so crazy about the clarity of distinction between 'aistate' and 'state' (from turn_state)-- I suppose your suggestion of shortening 'aistate' to 'ai' would be one way to deal with that, though I think I'd be a little more inclined to lengthen 'state' to 'turn_state' to strengthen the visual distinction

Reply:
```from aistate_interface import aistate``` is not possible without more sophisticated changes as the initialization currently happens after imports are finished.

So the choice is between 
```
from aistate_interface import get_aistate
get_aistate().do_stuff()
```

and something like
```
import aistate_interface as ai
ai.state.do_stuff()
```

The latter allows making stupid mistakes such as ```ai.state = 0``` or trying to import the state via the non-working ```from x import y``` statement. Should probably rename to ```_aistate``` when using the getter-approach to make that intent more clear.

Anyway, I have no strong preference.

Reply:
> So the choice is between

*cough* I will at least insist on comment 😄  on my suggestion of 
```
from aistate_interface import get_aistate

def some_func():
    aistate = get_aistate()
    aistate.do_stuff()
    aistate.do_other_stuff()
```

Reply:
Sorry if I gave you the impression I ignored you. I just focussed on the two possibilities as far as the interface itself is concerned. Usage in functions can of course happen as you suggested (eventhough it won't have any relevant performance impact except for very long loops).

Reply:
> from aistate_interface import aistate is not possible without more sophisticated changes as the initialization currently happens after imports are finished.

I missed that.

Please add this to getter docstring and prefix module variable with '_'.  This will be enough for this PR.

Everything possible, even code generation upon import.
https://plumbum.readthedocs.io/en/latest/local_commands.html#import-hack

Reply:
> though I think I'd be a little more inclined to lengthen 'state' to 'turn_state' to strengthen the visual distinction

Freeorion code are full of long lines.  `ai` and `state` may reduce this number, 'aistate' and `turn_state` may enlarge.   

Both pairs are meaningful enough fo me.  Long lines or lines with linebreaks impact readability.

Reply:
> computational cost of using a getter

Lets consider cost as zero and work with it in a way, that make code more readable in each context. 

`aistate` looks better as `ai_state`